### PR TITLE
[DROOLS-6786] Handle the case where a class is not found in JSONMarsh…

### PR DESCRIPTION
**Pull request by tkobayas**
…aller$CustomObjectDeserializer

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-6786

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java [hunk 1]
```diff
@@ -110,6 +110,15 @@
 
     private boolean findDeserializerFirst = Boolean.parseBoolean(System.getProperty("org.kie.server.json.findDeserializerFirst.enabled", "true"));
 
+    public static enum CNFEBehavior {
+        IGNORE,
+        WARN,
+        EXCEPTION
+    }
+
+    public static final String CUSTOM_OBJECT_DESERIALIZER_CNFE_BEHAVIOR = "org.kie.server.json.customObjectDeserializerCNFEBehavior";
```
[the remark this answers is not in the document]
Comment (tkobayas) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java]: @sutaakar Yes! Thanks for the suggestion.

**Comment by tkobayas** [comment]
Jenkins retest this

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java [hunk 2]
```diff
@@ -110,6 +109,14 @@
 
     private boolean findDeserializerFirst = Boolean.parseBoolean(System.getProperty("org.kie.server.json.findDeserializerFirst.enabled", "true"));
 
+    public static enum CNFEBehavior {
+        IGNORE,
+        WARN,
+        EXCEPTION
+    }
+
+    private CNFEBehavior customObjectDeserializerCNFEBehavior = CNFEBehavior.valueOf(System.getProperty(KieServerConstants.JSON_CUSTOM_OBJECT_DESERIALIZER_CNFE_BEHAVIOR, CNFEBehavior.IGNORE.name()));
```
Comment (sutaakar) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java:118]: Could you handle then case when user provides invalid system property value? Currently the produced error is not much descriptive.

Comment (tkobayas) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java]: @sutaakar Thanks, added.

**Review by sutaakar — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

**Pull request merged**